app/Controllers/Http/AuthController.ts

```diff
@@ -0,0 +1,52 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+
+import UsersService from "App/Services/user.service";
+import LoginValidator from 'App/Validators/LoginValidator';
+import RegisterUserValidator from "App/Validators/RegisterUserValidator";
+
+export default class AuthController {
+    public async registerUser({ auth, request, response }: HttpContextContract): Promise<any> {
+
+        try {
+            const payload = await request.validate(RegisterUserValidator);
```

Comment:
I think wrapping the code in try block is not required since the validate method already throws the error.